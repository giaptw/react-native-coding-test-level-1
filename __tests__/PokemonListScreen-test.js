import React from 'react';
import renderer from 'react-test-renderer';
import PokemonListScreen from '../src/screens/PokemonListScreen';

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn()
    }),
  };
});

jest.mock('react-redux', () => {
  const actualNav = jest.requireActual('react-redux');
  return {
    ...actualNav,
    useDispatch: () => ({}),
    useSelector: () => ({})
  };
});

test('renders correctly', () => {
  const tree = renderer.create(<PokemonListScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
