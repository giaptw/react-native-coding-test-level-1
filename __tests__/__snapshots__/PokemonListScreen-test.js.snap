// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 30,
        "justifyContent": "center",
        "paddingHorizontal": 20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": 30,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "height": "100%",
      }
    }
  >
    <RCTScrollView
      data={Object {}}
      getItem={[Function]}
      getItemCount={[Function]}
      initialNumToRender={10}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onEndReached={[Function]}
      onEndReachedThreshold={0}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={Array []}
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
